verilog/analysis/checkers/port_name_suffix_rule_test.cc

```diff
@@ -0,0 +1,106 @@
+      {"module t (input logic [7:0] name_i); endmodule;"},
+      {"module t (output logic [2:0] abc_o); endmodule;"},
+      {"module t (inout logic [3:0] xyz_io); endmodule;"},
+      {"module t (input bit name_i); endmodule;"},
+      {"module t (output bit abc_o); endmodule;"},
+      {"module t (inout bit xyz_io); endmodule;"},
+      {"module t (input logic name_i,\n"
+       "output logic abc_o,\n"
+       "inout logic xyz_io,\n"
+       "input logic [7:0] namea_i,\n"
+       "output logic [2:0] abca_o,\n"
+       "inout logic [3:0] xyza_io,\n"
+       "input bit nameb_i,\n"
+       "output bit abcb_o,\n"
+       "inout bit xyzb_io);\n"
+       "endmodule;"},
+  };
+  RunLintTestCases<VerilogAnalyzer, PortNameSuffixRule>(kTestCases);
+}
+
+// Tests that PortNameSuffixRule rejects invalid names.
+TEST(PortNameSuffixRuleTest, RejectTests) {
+  constexpr int kToken = SymbolIdentifier;
+  const std::initializer_list<LintTestCase> kTestCases = {
+      // General tests
+      {"module t (input logic ", {kToken, "name"}, "); endmodule;"},
+      {"module t (output logic ", {kToken, "abc"}, "); endmodule;"},
+      {"module t (inout logic ", {kToken, "xyz"}, "); endmodule;"},
+      {"module t (input logic [7:0] ", {kToken, "name"}, "); endmodule;"},
+      {"module t (output logic [2:0] ", {kToken, "abc"}, "); endmodule;"},
+      {"module t (inout logic [3:0] ", {kToken, "xyz"}, "); endmodule;"},
+      {"module t (input bit ", {kToken, "name"}, "); endmodule;"},
+      {"module t (output bit ", {kToken, "abc"}, "); endmodule;"},
+      {"module t (inout bit ", {kToken, "xyz"}, "); endmodule;"},
+      {"module t (input logic ", {kToken, "name"}, ",\n"
+       "output logic abc_o,\n"
+       "inout logic ", {kToken, "xyz"}, ",\n"
+       "input logic [7:0] namea_i,\n"
+       "output logic [2:0] ", {kToken, "abca"}, ",\n"
+       "inout logic [3:0] xyza_io,\n"
+       "input bit ", {kToken, "nameb"}, ",\n"
+       "output bit ", {kToken, "abcb"}, ",\n"
+       "inout bit xyzb_io);\n"
+       "endmodule;"},
+      // Invalid casing
+      {"module t (input logic ", {kToken, "name_I"}, "); endmodule;"},
+      {"module t (output logic ", {kToken, "abc_O"}, "); endmodule;"},
+      {"module t (inout logic ", {kToken, "xyz_IO"}, "); endmodule;"},
+      // no underscore
+      {"module t (input logic ", {kToken, "namei"}, "); endmodule;"},
+      {"module t (output logic ", {kToken, "abco"}, "); endmodule;"},
+      {"module t (inout logic ", {kToken, "xyzio"}, "); endmodule;"},
+      // Mismatched suffix tests
+      {"module t (input logic ", {kToken, "name_o"}, "); endmodule;"},
+      {"module t (input logic ", {kToken, "name_io"}, "); endmodule;"},
+      {"module t (output logic ", {kToken, "name_i"}, "); endmodule;"},
+      {"module t (output logic ", {kToken, "name_io"}, "); endmodule;"},
+      {"module t (inout logic ", {kToken, "name_i"}, "); endmodule;"},
+      {"module t (inout logic ", {kToken, "name_o"}, "); endmodule;"},
```

Comment:
suggest:

* `input` vs. `_no`
* `output` vs. `_ni`
* port declarations with (2+) multiple ids, _some_ or _all_ of which violate the naming guidance.